modules/hooks/server.lua

```diff
@@ -41,7 +41,7 @@ local function TriggerEventHooks(event, payload)
         for i = 1, #hooks do
 			local hook = hooks[i]
 
-			if hook.itemFilter and not itemFilter(hook.itemFilter, payload.fromSlot or payload.item or payload.itemName or payload.recipe, payload.toSlot) then
+			if hook.itemFilter and not itemFilter(hook.itemFilter, (event ~= 'buyItem' and payload.fromSlot) or payload.item or payload.itemName or payload.recipe, payload.toSlot) then
```

Comment:
Value of `fromSlot` is now a number, while itemFilter expects item name or item data table.
Because of this, buyItem's item filter was ignored.

This makes sure to ignore `fromSlot` value in it's item filter.

Reply:
Might as well check for value type, instead of event name. Let me know in that case.

Reply:
Should probably send the slot data?